### PR TITLE
fix edge case where position is closed before deleveraging

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
- technically it's possible that the position is closed before deleveraging is attempted. so this PR is meant to address that extreme edge case.

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/x/clob/keeper/liquidations.go [hunk 1]
```diff
@@ -138,8 +138,13 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
...
 		liquidationOrder := unfilledLiquidations[i]
 
-		subaccountId := liquidationOrder.GetSubaccountId()
 		perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
+		subaccountId := liquidationOrder.GetSubaccountId()
+
+		subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
+		if _, exists := subaccount.GetPerpetualPositionForId(perpetualId); !exists {
+			continue
+		}
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/liquidations.go:147]: Should we change [this line](https://github.com/dydxprotocol/v4-chain/blob/77573aaceca5bf83d9372b42b64ccf8f11e24dbf/protocol/x/clob/keeper/deleveraging.go#L56-L57) to a panic as part of this change (since now the position should always be open)?

Comment (lucas-dydx) [reply protocol/x/clob/keeper/liquidations.go:147]: Alternatively, wdyt of passing in `position` to `MaybeDeleverageSubaccount`? Up to you.

File: protocol/x/clob/keeper/liquidations.go [hunk 2]
```diff
@@ -138,8 +138,13 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 	for i := 0; i < int(k.Flags.MaxDeleveragingAttemptsPerBlock) && i < len(unfilledLiquidations); i++ {
 		liquidationOrder := unfilledLiquidations[i]
 
-		subaccountId := liquidationOrder.GetSubaccountId()
 		perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
+		subaccountId := liquidationOrder.GetSubaccountId()
+
+		subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
+		if _, exists := subaccount.GetPerpetualPositionForId(perpetualId); !exists {
+			continue
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/liquidations.go:146]: Is it worth statting how often this happens? Up to you, hopefully deleveraging is pretty rare so this doesn't happen.

**Review by lucas-dydx — APPROVED** [review]

**Review by lucas-dydx — APPROVED** [review]
